docker/volumes/db/webhooks.sql

```diff
@@ -25,81 +25,113 @@ BEGIN;
+            params = TG_ARGV[3]::jsonb;
+          END IF;
 
-      IF TG_ARGV[4] IS NULL OR TG_ARGV[4] = 'null' THEN
-        timeout_ms = 1000;
-      ELSE
-        timeout_ms = TG_ARGV[4]::integer;
-      END IF;
+          IF TG_ARGV[4] IS NOT NULL OR TG_ARGV[4] <> 'null' THEN
+            timeout_ms = TG_ARGV[4]::integer;
+          END IF;
 
-      CASE
-        WHEN method = 'GET' THEN
-          SELECT http_get INTO request_id FROM net.http_get(
-            url,
-            params,
-            headers,
-            timeout_ms
-          );
-        WHEN method = 'POST' THEN
-          payload = jsonb_build_object(
-            'old_record', OLD,
-            'record', NEW,
-            'type', TG_OP,
-            'table', TG_TABLE_NAME,
-            'schema', TG_TABLE_SCHEMA
-          );
+          -- Retry loop
+          WHILE NOT succeeded AND retry_count <= max_retries LOOP
+            PERFORM pg_sleep(retry_delays[retry_count+1]);
+            IF retry_delays[retry_count+1]>0 THEN
+               RAISE WARNING 'Retrying HTTP request: {retry_attempt: %, url: "%", timeout_ms: %, retry_delay_ms: %}',
+                 retry_count, url,timeout_ms,retry_delays[retry_count+1]*1000;
+            END IF;
+            retry_count := retry_count + 1;
+            BEGIN
+              CASE
+                WHEN method = 'GET' THEN
+                  SELECT http_get INTO local_request_id FROM net.http_get(
+                    url,
+                    params,
+                    headers,
+                    timeout_ms
+                  );
+                WHEN method = 'POST' THEN
+                  payload = jsonb_build_object(
+                    'old_record', OLD,
+                    'record', NEW,
+                    'type', TG_OP,
+                    'table', TG_TABLE_NAME,
+                    'schema', TG_TABLE_SCHEMA
+                  );
 
-          SELECT http_post INTO request_id FROM net.http_post(
-            url,
-            payload,
-            params,
-            headers,
-            timeout_ms
-          );
-        ELSE
-          RAISE EXCEPTION 'method argument % is invalid', method;
-      END CASE;
+                  SELECT http_post INTO local_request_id FROM net.http_post(
+                    url,
+                    payload,
+                    params,
+                    headers,
+                    timeout_ms
+                  );
+                ELSE
+                  RAISE EXCEPTION 'method argument % is invalid', method;
+              END CASE;
 
-      INSERT INTO supabase_functions.hooks
-        (hook_table_id, hook_name, request_id)
-      VALUES
-        (TG_RELID, TG_NAME, request_id);
-
-      RETURN NEW;
-    END
-  $function$;
+              IF local_request_id IS NOT NULL THEN
+                SELECT (response).status_code::integer
+                  INTO status_code
+                  FROM net._http_collect_response(local_request_id);
+                IF status_code < 400 THEN
```

Comment:
```suggestion
                IF status_code < 500 THEN
```

I think it's common to retry only 5xx errors. Because 4xx indicates a problem with client side request that typically can't be fixed by retry.

Perhaps we can also make it configurable with an array of retry-able status codes as argument.